server/app/repository/ApplicationEventRepository.java

```diff
@@ -0,0 +1,54 @@
+package repository;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+
+import com.google.common.collect.ImmutableList;
+import io.ebean.DB;
+import io.ebean.Database;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.Application;
+import models.ApplicationEvent;
+
+/**
+ * ApplicationEventRepository performs operations on {@link ApplicationEvent} that often involve
+ * other EBean models or asynchronous handling.
+ */
+public class ApplicationEventRepository {
```

Comment:
Please mark this as `final` to ensure that the decision to subclass is explicit rather than implicit.

Reply:
done.